src/ConsoleCommand.php

```diff
@@ -59,5 +73,21 @@ public function __unserialize(array $data): void
             objectOrMethod: $data['handler_class'],
             method: $data['handler_method'],
         );
+        $this->aliases = $data['aliases'];
+        $this->help = $data['help'];
+    }
+
+    /**
+     * @return array<string, ConsoleInputArgument>
+     */
+    public function getAvailableArguments(): array
+    {
+        $arguments = [];
+
+        foreach ($this->handler->getParameters() as $parameter) {
+            $arguments[$parameter->getName()] = ConsoleInputArgument::fromParameter($parameter);
```

Comment:
I would expect for this method to return some kind of definition, not a list of ConsoleInputArgument

Reply:
You could think of it like this: there are expected arguments/the definition, and then there are the actual values provided by the command. These are two separate concerns that should be split and matched together into something that can be passed to the handler. So I think we need at least three separate concepts here:

- An input parameter's definition
- The actual input value
- A class that matches these two together, doing all validation and possibly throwing exceptions if something's wrong
- The result (could be an array of keys/values) that can be given when the handler is invoked.